### DISCUSSION TITLE
Migration to a different appliance/Nextcloud installation

### DISCUSSION THREAD

**Discussion by ghost** [nextcloud/all-in-one#1492] [Questions] [answered]
The possibilities to switch to Nextcloud AIO are great but unfortunately I don't see a viable way to switch from Nextcloud AIO to other methods.

Maybe you could add the possibility to extport the database or similar solutions.

**Comment by szaimen** [comment, marked as answer]
Hi, as for any other Nextcloud installation, you can simply follow https://docs.nextcloud.com/server/stable/admin_manual/maintenance/migrating.html.

When you shutdown the containers from the AIO interface, you can find a database dump in the location `/var/lib/docker/volumes/nextcloud_aio_database_dump/_data`. 

The Nextcloud files are in `/var/lib/docker/volumes/nextcloud_aio_nextcloud/_data` and your private files/folders in `/var/lib/docker/volumes/nextcloud_aio_nextcloud_data/_data` or wherever you've chosen if you've changed your datadir location.

---

Apart from that there is still the user migration app.

**Reply by roib20** [reply]
Hello, I am attempting to restore docker volumes for a Nextcloud AIO installation. During initialization, I'm getting the error `DUMP dir is not writeable by postgres user.` from the `nextcloud-aio-database` container. This error is from [start.sh](https://github.com/nextcloud/all-in-one/blob/main/Containers/postgresql/start.sh).

I am assuming this is a permission issue that could be solved with `chmod` and `chown` on the host, similar to the instructions under `Access/Edit Nextcloud files/folders manually` in the README. However I am not sure what the correct permissions are to set for the database volume and all the other nextcloud-aio volumes. Do you happen to know the solution for this?

**Reply by szaimen** [reply]
Does this help?
https://github.com/nextcloud/all-in-one/blob/ab384f6e29567ecc306318c142c3697586e195f4/Containers/postgresql/Dockerfile#L11

**Reply by roib20** [reply]
I managed to get it to work by setting very permissive permissions for the host `/var/lib/docker/volumes`. I am aware this isn't a proper solution but anyway this isn't really an issue with nextcloud-aio, more me trying to test its limits alongside my unusual docker installation. This was just a test install of me seeing if I can restore a nextcloud-aio installation through backed-up named docker volumes. Regardless, your selected answer was very helpful so thank you.

**Reply by roib20** [reply]
Alright the command you linked actually was useful for figuring out which user ID I needed.

For future reference, the working commands (to run on the host) for setting permissions for the named volumes are below. **Please note** that running these commands is not normally necessary (unless facing permission/ownership issues like I did); in most cases docker can set the right permissions for volumes on its own.

For the nextcloud-aio-database named volume the commands are:
```
sudo chown -R 999:0 /var/lib/docker/volumes/nextcloud_aio_database/* && \
sudo chmod -R 750 /var/lib/docker/volumes/nextcloud_aio_database/*
```

And for the nextcloud-aio-nextcloud named volume the commands are:
```
sudo chown -R 33:0 /var/lib/docker/volumes/nextcloud_aio_nextcloud_data/* && \
sudo chmod -R 750 /var/lib/docker/volumes/nextcloud_aio_nextcloud_data/*
```

Other nextcloud-aio volumes seem to be less strict about which user owns them. By default docker volumes are owned by root.

**Comment by ghost** [comment]
Can this topic be included in the documentation? Especially with the description of:

"When you shut down the containers via the AIO interface, you will find a database dump in the /var/lib/docker/volumes/nextcloud_aio_database_dump/_data directory.

The Nextcloud files are in /var/lib/docker/volumes/nextcloud_aio_nextcloud/_data and your private files/folders are in /var/lib/docker/volumes/nextcloud_aio_nextcloud_data/_data or any other location you chose if you changed your data folder."

**Comment by ghost** [comment]
Is there a way to make the database available to log in to another computer on the network using DBeaver or pgAdmin?
